src/auth/authOS.ts

```diff
@@ -0,0 +1,212 @@
+import {useMutation, UseMutationOptions, useQuery, UseQueryOptions} from 'react-query'
+
+import {AUTH_METHOD_KEY, AUTH_METHOD_PIN, ENCRYPTED_PIN_HASH_KEY, useMutationWithInvalidations} from '../hooks'
+import {WalletMeta} from '../legacy/state'
+import storage from '../legacy/storage'
+import {Storage} from '../Storage'
+import {WalletJSON, walletManager, YoroiWallet} from '../yoroi-wallets'
+import {AuthMethod} from '../yoroi-wallets/types'
+import * as Keychain from './Keychain'
+
+export const useCanEnableAuthOs = (options?: UseQueryOptions<boolean, Error>) => {
+  const query = useQuery({
+    queryKey: ['canEnableAuthOs'],
+    queryFn: Keychain.canEnableAuthOs,
+    ...options,
+  })
+
+  return {
+    ...query,
+    canEnableOsAuth: query.data,
+  }
+}
+
+export const AUTH_METHOD_OS: AuthMethod = 'os'
+export const useEnableAuthWithOs = (
```

Comment:
is there a corresponding useDisableAuthWithOS?

Reply:
you only disable it, if you successfully create a PIN, so you disable it by `useEnableAuthWithPin` -> and vice versa.